distribution/src/main/resources/bin/elasticsearch-env.bat

```diff
@@ -16,6 +16,23 @@ for %%I in ("%ES_HOME%..") do set ES_HOME=%%~dpfI
 rem now set the classpath
 set ES_CLASSPATH=!ES_HOME!\lib\*
 
+rem check that JAVA_HOME does not contain qoutes
+if defined JAVA_HOME (
+  if "%JAVA_HOME:"=%" equ "" (
+    echo JAVA_HOME is empty. Specify a valid path for JAVA_HOME
+    exit /b 1
+  )
+
+  if '^%JAVA_HOME:~0^,1%=='^" (
+    echo JAVA_HOME cannot contain quotes ("). Remove the quotes from JAVA_HOME and try again.
+    exit /b 1
+  )
+  if '^%JAVA_HOME:~-1% == '^" (
```

Comment:
Perhaps merge these two if statements to one with e.g:

```cmd
SET JAVA_HOME_UNQUOTED=%JAVA_HOME:"=%
IF not "!foo!"=="!JAVA_HOME_UNQUOTED!" (
```

Since a double quote is [a reserved character in filenames on windows](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx?f=255&MSPPError=-2147217396)

Reply:
That makes things simpler :) 
Thanks!